lwm2m-client-ipso-digital-input.c

```diff
@@ -41,404 +41,262 @@
+    if (error != AwaError_Success)
+    {
+        printf("Failed to register flow access object\n");
+        return 1;
+    }
+
+    error = AwaStaticClient_DefineResourceWithHandler(awaClient, "State", IPSO_DIGITAL_INPUT_OBJECT, IPSO_DIGITAL_INPUT_STATE, AwaResourceType_Boolean, 0, 1, AwaResourceOperations_ReadOnly, digitalInputHandler);
+    if (error != AwaError_Success)
+    {
+        printf("Failed to define State resource\n");
+        return 1;
+    }
+
+    error = AwaStaticClient_DefineResourceWithHandler(awaClient, "Counter", IPSO_DIGITAL_INPUT_OBJECT, IPSO_DIGITAL_INPUT_COUNTER, AwaResourceType_Integer, 0, 1, AwaResourceOperations_ReadOnly, digitalInputHandler);
+    if (error != AwaError_Success)
+    {
+        printf("Failed to define Counter resource\n");
+        return 1;
+    }
+
+    error = AwaStaticClient_DefineResourceWithHandler(awaClient, "Polarity", IPSO_DIGITAL_INPUT_OBJECT, IPSO_DIGITAL_INPUT_POLARITY, AwaResourceType_Boolean, 0, 1, AwaResourceOperations_ReadWrite, digitalInputHandler);
+    if (error != AwaError_Success)
+    {
+        printf("Failed to define Polarity resource\n");
+        return 1;
+    }
+
+    error = AwaStaticClient_DefineResourceWithHandler(awaClient, "DebouncePeriod", IPSO_DIGITAL_INPUT_OBJECT, IPSO_DIGITAL_INPUT_DEBOUNCE_PERIOD, AwaResourceType_Time, 0, 1, AwaResourceOperations_ReadWrite, digitalInputHandler);
+    if (error != AwaError_Success)
+    {
+        printf("Failed to define DebouncePeriod resource\n");
+        return 1;
+    }
+
+    error = AwaStaticClient_DefineResourceWithHandler(awaClient, "EdgeSelection", IPSO_DIGITAL_INPUT_OBJECT, IPSO_DIGITAL_INPUT_EDGE_SELECTION, AwaResourceType_Integer, 0, 1, AwaResourceOperations_ReadWrite, digitalInputHandler);
+    if (error != AwaError_Success)
+    {
+        printf("Failed to define EdgeSelection resource\n");
+        return 1;
+    }
+
+    error = AwaStaticClient_DefineResourceWithHandler(awaClient, "ApplicationType", IPSO_DIGITAL_INPUT_OBJECT, IPSO_APPLICATION_TYPE, AwaResourceType_String, 0, 1, AwaResourceOperations_ReadOnly, digitalInputHandler);
+    if (error != AwaError_Success)
+    {
+        printf("Failed to define ApplicationType resource\n");
+        return 1;
+    }
+
+    error = AwaStaticClient_DefineResourceWithHandler(awaClient, "SensorType", IPSO_DIGITAL_INPUT_OBJECT , IPSO_SENSOR_TYPE, AwaResourceType_String, 0, 1, AwaResourceOperations_ReadOnly, digitalInputHandler);
+    if (error != AwaError_Success)
+    {
+        printf("Failed to define SensoryType resource\n");
+        return 1;
+    }
+
+    error = AwaStaticClient_DefineResourceWithHandler(awaClient, "CounterReset", IPSO_DIGITAL_INPUT_OBJECT , IPSO_DIGITAL_INPUT_COUNTER_RESET, AwaResourceType_Opaque, 0, 1, AwaResourceOperations_Execute, digitalInputHandler);
+    if (error != AwaError_Success)
+    {
+        printf("Failed to define CounterReset resource\n");
+        return 1;
+    }
```

Comment:
The IPSO digital input objects and their resources still need to be created after they are defined:

    for(int i = 0; i < DIGITAL_INPUTS; i++)
    {
        AwaStaticClient_CreateObject(awaClient, IPSO_DIGITAL_INPUT_OBJECT, i); //mandatory resources created here
        AwaStaticClient_CreateResource(awaClient, IPSO_DIGITAL_INPUT_OBJECT, i, IPSO_DIGITAL_INPUT_COUNTER);
    }